.travis.yml

```diff
@@ -1,36 +1,63 @@
 language: node_js
 matrix:
   fast_finish: true
-  # Generally using active LTS versions here, see https://github.com/nodejs/Release
+  # Generally using active LTS versions here, see https://github.com/nodejs/Release 
   include:
-    - name: "Test client"
-      node_js: 12
-      env: RUN="npm run test:client"
-    - name: "Test server"
-      node_js: 12
-      env: RUN="npm run test:server"
-    - name: "Lint and try building"
-      node_js: 12
-      env: RUN="npx concurrently --kill-others-on-fail 'npm:lint' 'npm:build'"
+  - name: Test client
+    node_js: 12
+    env: RUN="npm run test:client"
+  - name: Test server
+    node_js: 12
+    env: RUN="npm run test:server"
+  - name: Lint and build
+    node_js: 12
+    env: RUN="npx concurrently --kill-others-on-fail 'npm:lint' 'npm:build:tarball'" DEPLOY="true"
 # https://docs.travis-ci.com/user/reference/overview/#Virtualisation-Environment-vs-Operating-System
-script: $RUN
+script: "$RUN"
 install: npm ci
-dist: trusty
-services: mongodb
+dist: bionic
+services:
+  - mongodb
+  - docker
 addons:
   apt:
     packages:
     - graphicsmagick
 before_script:
-  - mongo --version
-  - mongod --version
-  # Ensure stable NPM version (avoids `-next.0` tags)
-  - 'npm install -g npm'
+- mongo --version
+- mongod --version
+# Ensure stable NPM version (avoids `-next.0` tags)
+- npm install -g npm
 cache:
   directories:
-    - "$HOME/.node-gyp"
-    - "$HOME/.npm"
-
+  - "$HOME/.node-gyp"
+  - "$HOME/.npm"
+# https://docs.travis-ci.com/user/deployment-v2
+before_deploy:
+  - sha256sum trustroots.tar.gz > trustroots.tar.gz.sha256
+deploy:
+  # this creates a tarball from the built files and uploads it to DigitalOcean spaces
+  # it is then available publically at https://trustroots-builds.fra1.digitaloceanspaces.com/trustroots.tar.gz
+  edge: true # opt in to use deployments v2
+  provider: s3
+  access_key_id: C6VPVZTPB2LWSGECOJAV
+  secret_access_key:
+    secure: KZ/M6Nnn3raktHWjYs0ddoRS/1ojfQpBXKiqroJMh1k2qy07j/GM5j/CKZH+aJGB5AU9QxKahF+zrQs2Gh5Uo3Uoo9gpVGUxqTb2Ok57zUBw3lywTbYCliisIpSA+NPLCzEI/C82Db603wk8yiD8TJyZCTPWFUyhGNUVKSy4Y/w=
+  bucket: trustroots-builds
+  endpoint: https://fra1.digitaloceanspaces.com
+  glob: "trustroots.tar.gz*"
+  acl: public_read
+  verbose: true
+  # https://docs.travis-ci.com/user/deployment-v2/conditional
+  on:
+    branch: master
+    condition: "$DEPLOY == 'true'" # only run for the build job
```

Comment:
> deploy section gets run on Travis only on merge, correct?

Yup, this is the bit that controls it:
- `master` branch only (to get it working in a branch/PR I had it running for all branches before)
- only if the $DEPLOY variable is set to true, so it only runs in the build matrix job...